### PR TITLE
ECS Web update tests

### PR DESCRIPTION
**Pull request by aguilinger**
Tests for updating ecs web. Note due to the deployment strategy, these tests do take quite a while. We probably will break them up in parallelization.

### CODE REVIEW

File: test/basic/ecs_web_update/ecs_web_update_test.go [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+			"container_name":           "nginx",
+			"is_public":                true,
+			"is_ecr_image":             false,
+			"vpc_name":                 "testecs-pub-to-priv-vpc",
+			"domain":                   "aptible-cloud-staging.com",
+			"subdomain":                "test-ecs-pub-integration",
+			"registry_credentials_arn": registrySecretArn,
+		},
+	})
+	defer cleanupAndAssert(t, terraformUpdateOptions)
```
Comment (madhuravius) [inline test/basic/ecs_web_update/ecs_web_update_test.go:207]: this is scary and i wouldn't do this. i think there's a few defers in here, there should only be one and i'd point it to the same set of options with a pointer if that's your desire (to have the final set), these will execute LIFO and it feels off

i'm down with doing this if you feel strongly but it feels weird

Comment (aguilinger) [reply test/basic/ecs_web_update/ecs_web_update_test.go:207]: I did remove this later

Comment (aguilinger) [reply test/basic/ecs_web_update/ecs_web_update_test.go:207]: But calling delete multiple times is fine, so I don't think this would cause issue?

Comment (madhuravius) [reply test/basic/ecs_web_update/ecs_web_update_test.go:207]: it should be idempotent, it just felt a bit off, as you could just reference a pointer that is mutating over the test suite and will pick up the latest vars

**Comment by madhuravius** [comment]
I think this one and the create are by far the hardest to test with the env varse/etc. 

This approach looks good but I would probably consider including some documentation soon (we should probably just do that as a group) when we unify the tests with a strategy

File: test/basic/ecs_web_update/ecs_web_update_test.go [hunk 2]
```diff
@@ -0,0 +1,707 @@
...
+		if *container.Name == "nginx" {
+			assert.Equal(t, "nginx", *container.Image)
+			assert.Equal(t, []*string{
+				aws.String("nginx"),
+				aws.String("-g"),
+				aws.String("daemon off;"),
+			}, container.Command)
+		}
+	}
+	assert.True(t, nginxFound)
```
Comment (madhuravius) [inline test/basic/ecs_web_update/ecs_web_update_test.go:581]: how does this pass 🤔

File: test/basic/ecs_web_update/ecs_web_update_test.go [hunk 3]
```diff
@@ -0,0 +1,707 @@
...
+		"vpc_name":          "testecs-secrets-vpc",
+		"domain":            "aptible-cloud-staging.com",
+		"subdomain":         "test-ecs-integration",
+	}
+	terraformOptions := terraform.WithDefaultRetryableErrors(t, &terraform.Options{
+		TerraformDir: ".",
+		Vars:         initialVariables,
+	})
+
+	// defer cleanupAndAssert(t, terraformOptions)
```
Comment (madhuravius) [inline test/basic/ecs_web_update/ecs_web_update_test.go:328]: uncomment prob

File: test/basic/ecs_web_update/ecs_web_update_test.go (hunk 2 above)
Comment (aguilinger) [reply test/basic/ecs_web_update/ecs_web_update_test.go:581]: this teaches me to not move around code without running the tests even if they take 30+ mins 😮‍💨

**Review by madhuravius — APPROVED** [review]

**Pull request merged**